scheds/include/scx/sdt_task_impl.bpf.h

```diff
@@ -392,6 +398,24 @@ static SDT_TASK_FN_ATTRS void sdt_task_free_idx(__u64 idx)
 	return;
 }
 
+static SDT_TASK_FN_ATTRS
+void __arena *sdt_task_retrieve(struct task_struct *p)
+{
+	struct sdt_task_data __arena *data;
+	struct sdt_task_map_val *mval;
+
+	sdt_arena_verify();
+
+	mval = bpf_task_storage_get(&sdt_task_map, p, 0, 0);
+	if (!mval)
+		return NULL;
+
+	data = mval->data;
+
+	return (void __arena *)data->payload;
```

Comment:
At some point was there a `cast_arena` helper? If not, it may be useful like the `cast_mask` for bpf  cpumasks.

Reply:
I don't think there was one, would it be used to ```const```ify the arena pointer? Or would it be like cast_{kern, user}() and transform a pointer back to an arena pointer?